torch/fx/experimental/proxy_tensor.py

```diff
@@ -470,6 +473,25 @@ def dispatch_trace(
     return GraphModule(tracer.root, graph, name)
 
 
+@contextlib.contextmanager
+def _pop_proxy_mode_temporarily(dk):
+    # This is a shim around the existng per-dispatch-key-mode logic.
+    # I'll delete the per-dispatch-key-mode logic in a followup PR
```

Comment:
Sorry, what's the plan here? It doesn't seem like this refactor eliminates the need for pre-dispatch modes?

Reply:
Are you going to make a dedicated pre-dispatch proxy slot?

Reply:
although - this might be worth revisiting if we eventually want functionalized pre-autograd tracing to work :)

Reply:
Yeah, I wouldn't delete the per-dispatch-key mode stack yet, we wanna use it for other stuff 📦